test/controllers/admin/partner_edit_site_test.rb

```diff
@@ -4,25 +4,131 @@
 
 class PartnerEditSiteTest < ActionDispatch::IntegrationTest
   setup do
+    Neighbourhood.destroy_all
```

Comment:
Basically minitest has these "fixtures" that are defined in test/fixtures that create a bunch of records in the DB before each run. Now our neighbourhood models don't have any kind of uniqueness test on their gritty details. So when tests are spun up these fixtures are hanging around. Now when we create a "partner", "site" or "user" etc these can also pop in a bunch of neighbourhoods as well.

then when it comes to doing a test like "give site neighbourhood and then make partner and give them an address in that neighbourhood" this test may run when you run the test in isolation but it may start failing if you're running it in a batch or if you just pushed to github and the CI is kicking off.

What is happening is that when you build a site a bunch of neighbourhoods and inserted and then when you create a partner another bunch of neighbourhoods are created. When you make an address for a partner it uses the postcode to lookup in postcode.io the ward it is in. (this query is stubbed in test/support/geocoder.rb). but the response it returns will be then looked up in our neighbourhoods table and linked to that neighbourhood.

but the neighbourhoods aren't unique! so you may be doing something perfectly but it will not appear on the site page!

so its easier just to nuke them all at the start.

i believe there is an issue about removing these fixtures and going through the factories but this is probably going to be a looooot of work.

Reply:
gotcha!